proxy/src/main/java/com/velocitypowered/proxy/connection/client/ConnectedPlayer.java

```diff
@@ -690,13 +690,17 @@ void teardown() {
       connectedServer.disconnect();
     }
 
-    Optional<Player> connectedPlayer = server.getPlayer(this.getUniqueId());
     server.unregisterConnection(this);
 
     DisconnectEvent.LoginStatus status;
+    Optional<Player> connectedPlayer = server.getPlayer(this.getUniqueId());
     if (connectedPlayer.isPresent()) {
```

Comment:
This is still incorrect. Now the successful login case is never fired. Your first revision was correct.

To fix this just put this line back where it was - before the unregisterConnection call.

Reply:
I commented on the wrong line, the offending line is above this one. I was reviewing this from my phone. Whoops!

Reply:
Should be fixed now.